mempool/mempool.go

```diff
@@ -256,12 +266,33 @@ func (mem *Mempool) resCbRecheck(req *abci.Request, res *abci.Response) {
 			// Done!
 			atomic.StoreInt32(&mem.rechecking, 0)
 			mem.logger.Info("Done rechecking txs")
+
+			mem.notifyIfTxsAvailable()
 		}
 	default:
 		// ignore other messages
 	}
 }
 
+// TxsAvailable returns a channel which fires once for every height,
+// and only when transactions are available in the mempool.
+// NOTE: the returned channel may be nil if EnableTxsAvailable was not called.
+func (mem *Mempool) TxsAvailable() chan int {
+	return mem.txsAvailable
+}
+
+func (mem *Mempool) notifyIfTxsAvailable() {
```

Comment:
s/notifyIfTxsAvailable/notifyTxsAvailable/g

We're notifying that txs are available.  The conditional inside the function is whether we care that txs are available.  The caller ensures that this function is only called when txs are available.